### PR TITLE
NO-SNOW: Expose the Ingest SDK MAX_CLIENT_LAG configuration in KC 

### PR DESCRIPTION
**Pull request by sfc-gh-tzhang**
When schematization is enabled, we're inserting row by row in order to preserve the original order of the batch, this leads to an issue if the data validation for each row is slow, we might end up with many small files since the SDK flushes every second. Exposing the MAX_CLIENT_LAG helps solve this issue shoer term. Longer term, it would be good if we can avoid inserting row by row for schematization.

- Support a new configuration `snowflake.streaming.max.client.lag` in KC. Instead of adding a new configuration, i update the blob version config since we don't support ARROW in the new SDK, adding less code is always better
- Extend the `StreamingClientProperties` to use different client if the `parameterOverrides` is also different
- Corresponding test cases added

### CODE REVIEW

**Review by sfc-gh-rcheng — APPROVED** [review]
lgtm! concerned about if we need to notify customers or change docs if the bdec version override parameter is removed

File: src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java [hunk 1]
```diff
@@ -1461,12 +1462,49 @@ public void testSchematizationSchemaEvolutionWithNonNullableColumn() throws Exce
...
+      TestUtils.assertWithRetry(
+          () -> service.getOffset(new TopicPartition(topic, partition)) == noOfRecords, 5, 4);
+      Assert.fail("The rows should not be flushed");
+    } catch (Exception e) {
+      // do nothing
+    }
+
+    // Wait for enough time, the rows should be flushed
+    TestUtils.assertWithRetry(
+        () -> service.getOffset(new TopicPartition(topic, partition)) == noOfRecords, 30, 6);
```
Comment (sfc-gh-rcheng) [inline src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:1496]: your github actions test is failing on this, probably just need to bump the retry count?

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java [hunk 2]
```diff
@@ -117,9 +117,9 @@ public class SnowflakeSinkConnectorConfig {
   public static final String INGESTION_METHOD_DEFAULT_SNOWPIPE =
       IngestionMethodConfig.SNOWPIPE.toString();
 
-  // This is the streaming bdec file version which can be defined in config
-  // NOTE: Please do not override this value unless recommended from snowflake
```
[the remark this answers is not in the document]
Comment (sfc-gh-tjones) [reply src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java:121]: This was meant to be internal only so I think it's fine to remove. I believe we track old file versions in Snowflake and alert if we detect arrow files.

File: src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java (hunk 1 above)
Comment (sfc-gh-tzhang) [reply src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:1496]: Yeah, I don't know why, I think 180 seconds should be enough....

Comment (sfc-gh-rcheng) [reply src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:1496]: I've similar issues (passes locally, consistently fails on github) when i used 100 seconds, had to bump the total retry to 400 seconds

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java [hunk 3]
```diff
@@ -77,15 +78,15 @@ public StreamingClientProperties(Map<String, String> connectorConfig) {
         STREAMING_CLIENT_PREFIX_NAME
             + connectorConfig.getOrDefault(Utils.NAME, DEFAULT_CLIENT_NAME);
 
-    // Override only if bdec version is explicitly set in config, default to the version set
-    // inside Ingest SDK
+    // Override only if the max client lag is explicitly set in config
     this.parameterOverrides = new HashMap<>();
     Optional<String> snowpipeStreamingBdecVersion =
```
Comment (sfc-gh-rcheng) [inline src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java:83]: change: snowpipeStreamingBdecVersion -> snowpipeStreamingMaxClientLag or something similar

File: src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java (hunk 1 above)
Comment (sfc-gh-tzhang) [reply src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java]: ok, bumping again...seems very weird, this is something I want to investigate after coming back

**Comment by sfc-gh-rcheng** [comment]
merging for toby - needed to create securonix jar

**Pull request merged**